src/coffee/browser/browser.coffee

```diff
@@ -71,4 +72,71 @@ angular.module 'app.browser', []
   $scope.$on '$destroy', ->
     $interval.cancel bgCycler
 
-  return 
+  # Handle drag and drop event
+  angular.element($window).bind 'dragenter', (event) ->
+    event.preventDefault()
+    event.stopPropagation()
+    false
+  angular.element($window).bind 'dragover', (event) ->
+    event.preventDefault()
+    event.stopPropagation()
+    false
+  angular.element($window).bind 'drop', (event) ->
+
+    if event.handled != true
+
+      event.preventDefault()
+      event.stopPropagation()
+
+      # Only take the first file in case there is multiple
+      file = event.dataTransfer.files[0]
+      reader = new FileReader
+      reader.readAsText file
+
+      # From here we need to do several things
+      # 1. Check wether the file is a torrent file
+      # 2. If it does -> play the torrent
+      # 3. If it doesn't -> display a warning message to the user
+      reader.onload = ((file) ->
+        (e) ->
+          console.log file.name
+          console.log file.type
+          console.log file.size
+          console.log file.lastModifiedDate
+          # console.log reader.result
+          return
+      )(file)
+
+    else
+      event.handled = true
+
+    false
+
+  # Handle Ctrl+V to paste magnet link
+  $scope.ctrlDown = false
```

Comment:
Use vm. Instead of $scope as $scope is depreciated in angular 2 

Eg in the html it would be {{ browser.ctrlDown }}